src/main/java/frc/robot/subsystems/swerve/Swerve.java

```diff
@@ -93,8 +93,22 @@ public Swerve(boolean isReal) {
       try (Notifier odometryUpdater =
           new Notifier(
               () -> {
-                odometry.updateWithTime(
-                    Timer.getFPGATimestamp(), gyro.getRotation2d(), getModulePositions());
+                try {
+                  modules[0].periodic();
+                  modules[1].periodic();
+                  modules[2].periodic();
+                  modules[3].periodic();
+
+                  odometry.updateWithTime(
+                      Timer.getFPGATimestamp(),
+                      (DriverStation.getAlliance().get() == DriverStation.Alliance.Blue)
```

Comment:
Why does the robot pose depend on the alliance? The field is fixed and the robot pose should be the same whether it is on red or blue alliance.

Reply:
The field is fixed but the zero angle of the gyro isn't